frontend/src/utils/StatusUtils.ts

```diff
@@ -26,4 +26,19 @@ function getStatusVariant(status: string | undefined) {
   }
 }
 
-export { convertStatusTitleCase, getStatusVariant };
+function getLevelVariant(level: number) {
```

Comment:
nit: we dont _really_ need a full variant for each level. We only set the `background`. Instead, make `getBadgeColorByLevel` and return strings like `"purple.50"`. Then you can do `<Badge background={getBadgeColorByLevel(level)} size="s">` and you can eliminate the variant changes to frontend/src/theme/components/Badge.ts